modules/apps/bean-portlet/bean-portlet-cdi-extension/src/main/java/com/liferay/bean/portlet/cdi/extension/internal/scope/ScopedBeanManagerThreadLocal.java

```diff
@@ -78,8 +78,12 @@ public static Closeable install(ScopedBeanManager scopedBeanManager) {
 			unsafeRunnable.run();
 		}
 		finally {
-			if (empty) {
-				scopedBeanManagers.pop();
+			if (!scopedBeanManagers.isEmpty()) {
```

Comment:
@ngriffin7a can scopedBeanManagers be empty here? You just made the push unconditional in the previous commit, won't that make it impossible for scopedBeanManagers to be empty here? Is any unsafeRunnable here messing with the threadlocal scopedBeanManagers ?

Reply:
@shuyangzhou Good catch. The check for `if (!scopedBeanManagers.isEmpty()) {` on line 81 is unnecessary. I removed it locally and tested the V3ResourceAsyncTests from the Portlet 3.0 TCK and it worked. I also checked in the debugger and the stack is always non-empty.